src/mutation.cpp

```diff
@@ -1048,6 +1088,14 @@ void Character::mutate( const int &true_random_chance, bool use_vitamins )
         cat = *cat_list.pick();
         cat_list.add_or_replace( cat, 0 );
         add_msg_debug( debugmode::DF_MUTATION, "Picked category %s", cat.c_str() );
+        // only decide if it's good or bad after we pick the category
+        if( roll_bad_mutation( cat ) ) {
+            // If we picked bad, mutation can be bad or neutral
+            allow_bad = true;
+        } else {
+            // Otherwise, can be good or neutral
```

Comment:
```suggestion
        // Only decide if it's good or bad after we pick the category.
        if( roll_bad_mutation( cat ) ) {
            // If we picked bad, mutation can be bad or neutral.
            allow_bad = true;
        } else {
            // Otherwise, can be good or neutral.
```